### PR TITLE
klient/machine: add helper function that checks if two index meta changes are similar

### PR DESCRIPTION
**Pull request by ppknap**
## Motivation and Context
Detect and split changes for cases like "add- modify - delete" chain of file events.

## How Has This Been Tested?
Unit tests

### CODE REVIEW

File: go/src/koding/klient/machine/index/change_test.go [hunk 1]
```diff
@@ -229,7 +229,62 @@ func TestChangeCoalesceConcurrent(t *testing.T) {
...
+		},
+		"U is UL": {
+			A:      index.ChangeMetaUpdate,
+			B:      index.ChangeMetaUpdate | index.ChangeMetaLocal,
+			Result: true,
+		},
+		"UL is U": {
+			A:      index.ChangeMetaUpdate | index.ChangeMetaLocal,
+			B:      index.ChangeMetaUpdate,
+			Result: true,
```
Comment (rjeczalik) [inline go/src/koding/klient/machine/index/change_test.go]: If no direction is specified then local is implicitly assumed, right?

**Review by rjeczalik — APPROVED** [review]

File: go/src/koding/klient/machine/index/change_test.go (hunk 1 above)
Comment (ppknap) [inline go/src/koding/klient/machine/index/change_test.go]: you're correct

**Comment by ppknap** [comment]
@cihangir gentle ping

**Review by cihangir — APPROVED** [review]
💯

**Pull request merged**